pkg/gatherers/clusterconfig/gather_monitoring_pvs.go

```diff
@@ -0,0 +1,142 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// GatherMonitoringPVs Collects Persistent Volumes from openshift-monitoring namespace
+// which matches with ConfigMap configuration yaml
+//
+// ### API Reference
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/configmap.go
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/persistentvolume.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/persistentvolumes/pvc-{hash-id}.json
+//
+// ### Location in archive
+// - `config/persistentvolumes/{persistent_volume_name}.json`
+//
+// ### Config ID
+// `clusterconfig/monitoring_persistent_volumes`
+//
+// ### Released version
+// - 4.13
+//
+// ### Backported versions (TODO: update backported releases)
+// - 4.12
+// - 4.11
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherMonitoringPVs(ctx context.Context) ([]record.Record, []error) {
+	kubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	mg := MonitoringPVGatherer{ctx: ctx, client: kubeClient.CoreV1()}
+
+	name, errors := mg.getDefaultPrometheusName()
+	if len(errors) > 0 {
+		return nil, errors
+	}
+
+	fmt.Printf("name: %v\n", name)
+
+	return mg.gather(name)
+}
+
+type MonitoringPVGatherer struct {
+	ctx    context.Context
```

Comment:
This (storing context in structs) is considered a bad practice - check https://pkg.go.dev/context

Reply:
aw man. They explicitly say it
> Do not store Contexts inside a struct type;
LOL, my bad. I'll fix this and keep it on my memory.
Thanks.